docs/examples/encrypt_and_transfer.rst

```diff
@@ -1,4 +1,79 @@
+
+
+Steps
+-----
+We start by writing our own custom tool. For the example, MakeFiles is a trivial tool that creates three files in a given folder.
+The funcx function that achieves this functionality is ``makeFiles()``. We then define the tool using ``GladierBaseTool``. 
+
+
+.. code-block:: python
+    
+    @generate_flow_definition
+    class MakeFiles(GladierBaseTool):
+        funcx_functions = [makeFiles]
+        required_input = [
+            'make_input',
+            'funcx_endpoint_compute'
+        ]
+
+Defining a workflow is similar to the previous case, where we override the ``GladierBaseClient`` class. In this case, we want to use the MakeFiles, Tar, Encrypt, and Transfer tools.
+
+
+.. code-block:: python
+
+    @generate_flow_definition
+    class CustomTransfer(GladierBaseClient):
+        gladier_tools = [
+            MakeFiles,
+            'gladier_tools.posix.Tar',
+            'gladier_tools.posix.Encrypt',
+            'gladier_tools.globus.Transfer',
+        ]
+    
+The ``@generate_flow_definition`` decorator takes the flow definitions of the individual tools and constructs a flow definition for the new flow, so we do not have 
+to define a custom flow definition. To view the constructed flow definition, use ``pprint(ct.flow_definition)``. Refer to the `Flow Generation <https://gladier.readthedocs.io/en/docs/flow_generation.html>`_ doc for more details on the same.
+
+The next step is to define the input for the flow. It might be helpful to refer to the docs for each of the tools to find out what needs to be passed in as input. 
+For example, here are the documentations for the `Tar <https://gladier.readthedocs.io/en/docs/tools/posix/tar.html>`_, `Encrypt <https://gladier.readthedocs.io/en/docs/tools/posix/encrypt.html>`_ and `Transfer <https://gladier.readthedocs.io/en/docs/tools/globus/transfer.html>`_ tools. Feel free to use the below blueprint:
+
+.. code-block:: python
+    
+    flow_input = {
+        'input': {
+            # Set this to the folder in which you want to run the makeFiles function in
+            'make_input': '',
+            # Set this to the same folder as above
+            'tar_input': '',
+            # Set this to the resultant archive of the above folder
+            'encrypt_input': '',
+            # Set this to the symmetric key you want to use to encrypt/decrypt the file
+            'encrypt_key': '',
+            # Set this to your own funcx endpoint where you want to encrypt files
+            'funcx_endpoint_compute': '',
+            # Set this to the globus endpoint where your encrypted archive has been created
+            'transfer_source_endpoint_id': '',
+            # By default, this will transfer the encrypt file to Globus Tutorial Endpoint 1
+            'transfer_destination_endpoint_id': 'ddb59aef-6d04-11e5-ba46-22000b92c6ec',
+            'transfer_source_path': '',
+            'transfer_destination_path': '',
+            'transfer_recursive': False,
```

Comment:
Are there some good defaults we could add to this, so the user can copy/paste/run without any changes?

Reply:
I used the /tmp/myfiles and kept that going, so transfer_source_path became /tmp/myfiles.tgz.aes

Reply:
That sounds good